Assets/Scripts/Singletons/PilotNameManager.cs

```diff
@@ -0,0 +1,134 @@
+﻿using System.Collections.Generic;
+using System.Threading.Tasks;
+using UnityEngine;
+
+public class PilotNameManager : MonoBehaviour
+{
+	public static PilotNameManager Instance { get; private set; }
+	public string[] HumanMaleNames { get; private set; }
+	public string[] HumanFemaleNames { get; private set; }
+	public string[] HelicidNames { get; private set; }
+	public string[] OshunianNames { get; private set; }
+	public string[] OshunianTitles { get; private set; }
+	public string[] VestaPrefixes { get; private set; }
+	public string[] VestaNames { get; private set; }
+
+	private readonly int robotPrefixLength = 3;
+	private readonly int robotSuffixLength = 4;
+
+	/// <summary>
+	/// Species name formats:
+	///		Human: [FirstName] [Initial].
+	///		Helicid: [LastName] [Initial]. 
+	///		Oshunian: [FirstName] [Title]
+	///		Vesta: [Prefix-[Name]
+	///		Robot: [Alphabetical prefix]-[Numerical suffix]
+	/// </summary>
+
+	private static System.Random random;
+
+	private void Awake()
+	{
+		if (Instance == null)
+		{
+			Instance = this;
+			random = new System.Random();
+			DontDestroyOnLoad(gameObject);
+        }
+		else
+		{
+			Destroy(gameObject);
+			return;
+		}
+	}
+
+	public async void Init()
+	{
+        List<Task> pilotNameTasks = new List<Task>
+        {
+            Task.Factory.StartNew(() => HumanMaleNames = LoadNamePoolAsync(PilotsConstants.humanMaleNamesPath).Result),
+            Task.Factory.StartNew(() => HumanFemaleNames = LoadNamePoolAsync(PilotsConstants.humanFemaleNamesPath).Result),
+            Task.Factory.StartNew(() => HelicidNames = LoadNamePoolAsync(PilotsConstants.helicidNamesPath).Result),
+            Task.Factory.StartNew(() => OshunianNames = LoadNamePoolAsync(PilotsConstants.oshunianNamesPath).Result),
+            Task.Factory.StartNew(() => OshunianTitles = LoadNamePoolAsync(PilotsConstants.oshunianTitlesPath).Result),
+            Task.Factory.StartNew(() => VestaPrefixes = LoadNamePoolAsync(PilotsConstants.vestaPrefixesPath).Result),
+            Task.Factory.StartNew(() => VestaNames = LoadNamePoolAsync(PilotsConstants.vestaNamesPath).Result)
+        };
+		await Task.WhenAll(pilotNameTasks).ContinueWith(x => PilotsManager.Instance.RandomisePilots());

```

Comment:
There's probably better ways to do this, but I had trouble loading the name pools async before the pilot manager method call

Reply:
this works for now